cpp/include/cudf/scalar/scalar_factories.hpp

```diff
@@ -121,6 +121,20 @@ std::unique_ptr<scalar> make_default_constructed_scalar(
   rmm::cuda_stream_view stream        = rmm::cuda_stream_default,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Creates an empty (invalid) scalar of the same type as the `input` column_view.
+ *
+ * @throw std::bad_alloc if device memory allocation fails
```

Comment:
```suggestion
@throw cudf::logic_error if the `input` column is struct type and empty
```
The `bad_alloc` is not thrown explicitly inside this function.